moveit_ros/moveit_servo/src/servo_calcs.cpp

```diff
@@ -294,17 +296,19 @@ void ServoCalcs::calculateSingleIteration()
   have_nonzero_twist_stamped_ = latest_nonzero_twist_stamped_;
   have_nonzero_joint_command_ = latest_nonzero_joint_cmd_;
 
+  planning_scene_monitor_->requestPlanningSceneState();
```

Comment:
Why keep requesting the whole planning scene on each iteration .? see https://github.com/ros-planning/moveit2/pull/673/files for how we fixed this issue for moveit2

Reply:
I'm don't know how to do that, can you elaborate a bit more?

Reply:
I believe @JafarAbdi means, you only have to do this once, during initialization. Since it is already done at L197, you don't need to do it again here. (Please double-check that this works, though. I'm not 100% sure.)

```suggestion
```

Reply:
This doesn't work. See my other comment [here](https://github.com/ros-planning/moveit/pull/2894#issuecomment-934144401)